File: create-icons.py
#!/usr/bin/env python3
"""
Create PNG icons for Chrome extension
"""

from PIL import Image, ImageDraw, ImageFont
import os

def create_icon(size):
    """Create a square icon with gradient background and text"""
    # Create image with RGBA mode
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Create gradient background
    for y in range(size):
        # Calculate gradient color
        ratio = y / size
        r = int(59 + (139 - 59) * ratio)
        g = int(130 + (92 - 130) * ratio)
        b = int(246 + (246 - 246) * ratio)
        
        draw.line([(0, y), (size, y)], fill=(r, g, b, 255))
    
    # Draw text
    try:
        # Try to use a font
        font_size = int(size * 0.6)
        font = ImageFont.truetype("arial.ttf", font_size)
    except:
        # Fall back to default font
        font = ImageFont.load_default()
    
    # Get text size
    text = "S"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    # Center text
    x = (size - text_width) // 2
    y = (size - text_height) // 2
    
    draw.text((x, y), text, fill=(255, 255, 255, 255), font=font)
    
    return img

def main():
    """Create icons in different sizes"""
    sizes = [16, 32, 48, 128]
    
    # Create output directory
    output_dir = "browser-extension-store/icons"
    os.makedirs(output_dir, exist_ok=True)
    
    for size in sizes:
        print(f"Creating {size}x{size} icon...")
        icon = create_icon(size)
        output_path = os.path.join(output_dir, f"icon{size}.png")
        icon.save(output_path, "PNG")
        print(f"Saved: {output_path}")
    
    print("All icons created successfully!")

if __name__ == "__main__":
    try:
        from PIL import Image, ImageDraw, ImageFont
        main()
    except ImportError:
        print("PIL library not found. Installing...")
        import subprocess
        import sys
        
        # Install PIL
        subprocess.check_call([sys.executable, "-m", "pip", "install", "Pillow"])
        
        # Retry
        from PIL import Image, ImageDraw, ImageFont
        main()
    except Exception as e:
        print(f"Error creating icons: {e}")
        print("Please create PNG icons manually using an image editor.")
        print("Required sizes: 16x16, 32x32, 48x48, 128x128")
        print("Design: Blue gradient background with white 'S' text")

File: create-icons.cs
using System;
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Drawing.Text;
using System.IO;

class Program
{
    static void Main()
    {
        // 创建不同尺寸的图标
        int[] sizes = { 16, 32, 48, 128 };
        
        foreach (int size in sizes)
        {
            CreateIcon(size);
        }
        
        Console.WriteLine("PNG icons created successfully!");
    }
    
    static void CreateIcon(int size)
    {
        // 创建位图
        using (Bitmap bitmap = new Bitmap(size, size))
        using (Graphics graphics = Graphics.FromImage(bitmap))
        {
            // 设置高质量渲染
            graphics.SmoothingMode = SmoothingMode.AntiAlias;
            graphics.TextRenderingHint = TextRenderingHint.AntiAlias;
            graphics.CompositingQuality = CompositingQuality.HighQuality;
            
            // 创建渐变背景
            using (LinearGradientBrush brush = new LinearGradientBrush(
                new Rectangle(0, 0, size, size),
                ColorTranslator.FromHtml("#3b82f6"),
                ColorTranslator.FromHtml("#8b5cf6"),
                LinearGradientMode.ForwardDiagonal))
            {
                graphics.FillRectangle(brush, 0, 0, size, size);
            }
            
            // 计算圆角半径
            int cornerRadius = (int)(size * 0.2);
            
            // 创建圆角路径
            using (GraphicsPath path = new GraphicsPath())
            {
                path.AddArc(0, 0, cornerRadius * 2, cornerRadius * 2, 180, 90);
                path.AddArc(size - cornerRadius * 2, 0, cornerRadius * 2, cornerRadius * 2, 270, 90);
                path.AddArc(size - cornerRadius * 2, size - cornerRadius * 2, cornerRadius * 2, cornerRadius * 2, 0, 90);
                path.AddArc(0, size - cornerRadius * 2, cornerRadius * 2, cornerRadius * 2, 90, 90);
                path.CloseAllFigures();
                
                // 应用圆角遮罩
                using (Region region = new Region(bitmap.GetBounds()))
                {
                    region.Intersect(path);
                    graphics.SetClip(region, CombineMode.Replace);
                    
                    // 重新绘制渐变背景
                    using (LinearGradientBrush brush = new LinearGradientBrush(
                        new Rectangle(0, 0, size, size),
                        ColorTranslator.FromHtml("#3b82f6"),
                        ColorTranslator.FromHtml("#8b5cf6"),
                        LinearGradientMode.ForwardDiagonal))
                    {
                        graphics.FillRectangle(brush, 0, 0, size, size);
                    }
                }
            }
            
            // 绘制文字
            using (Font font = new Font("Arial", (int)(size * 0.6), FontStyle.Bold))
            using (SolidBrush textBrush = new SolidBrush(Color.White))
            using (StringFormat stringFormat = new StringFormat()
            {
                Alignment = StringAlignment.Center,
                LineAlignment = StringAlignment.Center
            })
            {
                graphics.DrawString("S", font, textBrush, size / 2f, size / 2f, stringFormat);
            }
            
            // 保存为PNG
            string outputPath = Path.Combine(Directory.GetCurrentDirectory(), "browser-extension-store", "icons", $"icon{size}.png");
            bitmap.Save(outputPath, System.Drawing.Imaging.ImageFormat.Png);
            
            Console.WriteLine($"Created: {outputPath}");
        }
    }
}